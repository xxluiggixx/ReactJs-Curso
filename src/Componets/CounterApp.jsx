import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ( { value } ) =>{

    const [count, setCount] = useState(value)
    
    const increase = ()=> setCount(count+1)
    const decrease = ()=> setCount(count-1)
    const reset = ()=> setCount(value)

return(
    <>
    <h1>CounterApp</h1>
    <h2> { count } </h2>
    <button onClick={ increase }>{ '+1' }</button>
    <button onClick={ reset }>{ 'Reset' }</button>
    <button onClick={ decrease }>{ '-1' }</button>
    </>
)
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}