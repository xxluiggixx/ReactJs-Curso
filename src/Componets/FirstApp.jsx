const newMsg = () => {
    return "Hola mundo!"
}


const FirstApp = () => {
    return(
        <h2>{ newMsg() }</h2>
    )
}

export default FirstApp;