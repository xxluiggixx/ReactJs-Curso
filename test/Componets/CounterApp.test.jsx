import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../../src/Componets/CounterApp";
/*  
test: Debe de hacer match con el snapshot
Test: debe de mostar el valor inicial de 100 <CounterApp value={100}/> 
*/
describe('Test on file CounterApp', () => { 
    const initialValue= 100
    test('should match with snapshot', () => { 
        const { container } = render(<CounterApp value={initialValue}/> )
        expect(container).toMatchSnapshot();
     })
     test('should match contain value 100', () => { 
        render(<CounterApp value={initialValue}/> )
        expect(screen.getByText(initialValue)).toBeTruthy();
     })
     test('should Increase with button +1', () => { 
        render(<CounterApp value={initialValue}/> );
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('101')).toBeTruthy();
      })
     test('should Decrement with button -1', () => { 
        render(<CounterApp value={initialValue}/> );
        fireEvent.click( screen.getByText('-1') )
        expect( screen.getByText('99')).toBeTruthy();
      })
     test('should reset with button Reset', () => { 
        render(<CounterApp value={initialValue}/> );
        fireEvent.click( screen.getByText('-1') )
        fireEvent.click( screen.getByText('-1') )
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );
        expect( screen.getByText( initialValue ) ).toBeTruthy();
      })
 })