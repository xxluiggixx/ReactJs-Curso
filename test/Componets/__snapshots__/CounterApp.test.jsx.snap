// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file CounterApp should match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
    -1
  </button>
</div>
`;
